server/api/authentication.js

```diff
@@ -0,0 +1,156 @@
+const router = require('express').Router();
+const brcypt = require('bcrypt');
+const hash = require('../hash');
+const { User, Session, Order } = require('../db');
+
+const A_WEEK_IN_SECONDS = 1000 * 60 * 60 * 24 * 7;
+
+router.post('/mount', async (req, res, next) => {
+  try {
+    if (req.session) {
+      const refreshedSession = await Session.findByPk(req.session.id);
+      res.cookie('sid', refreshedSession.uuid, {
+        maxAge: A_WEEK_IN_SECONDS,
+        path: '/',
+      }).send(refreshedSession);
+    } else {
+      const guestSession = await Session.create();
+      res.cookie('sid', guestSession.uuid, {
+        maxAge: A_WEEK_IN_SECONDS,
+        path: '/',
+      }).status(201).send(guestSession);
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.post('/login', async (req, res, next) => {
+  const { username, password } = req.body;
+  if (typeof username !== 'string' || typeof password !== 'string') {
+    res.status(400).send({
+      message: 'Username and password must both be strings.',
+    });
+  } else {
+    try {
+      const user = await User.findOne({
+        where: {
+          username,
+        },
+        include: [Session],
+      });
+
+      if (user) {
+        const comparisonResult = await brcypt.compare(password, user.password);
+        if (!comparisonResult) {
+          throw new Error('Wrong password!');
+        }
+        if (user.session) {
+          res.cookie('sid', user.session.uuid, {
+            maxAge: A_WEEK_IN_SECONDS,
+            path: '/',
+          });
+
+          const guestSession = req.session;
+          const guestCart = await Order.findOne({
+            where: {
+              sessionId: guestSession.id,
+              isPaid: false,
+            },
+          });
+          if (guestCart) {
+            const existingUserCart = await Order.findOne({
+              where: {
+                sessionId: user.session.id,
+                isPaid: false,
+              },
+            });
+            if (existingUserCart) {
+              const guestCartCostumes = await guestCart.getCostumes();
+              await existingUserCart.addCostumes(guestCartCostumes);
+              await existingUserCart.calcTotal();
+              await guestCart.destroy();
+            } else await guestCart.setSession(user.session);
+          }
+          await guestSession.destroy();
```

Comment:
This is likely causing bugs? We either want to associate the now logged in user with this session, or create a new session with this user.